### PR TITLE
Release 2.0.0

### PR DESCRIPTION
**Pull request by ygoyal18**
# Changelog

### Added
* [mbed-edge] Upgraded mbed-edge, also known as edge-core, from 0.8.0 to 0.12.0.
* [maestro] Fixed DHCP client in maestro daemon to reorder options and parameters and to add right options to discover packet.
* [maestro] Running maestro process with Go env flag - `GODEBUG=madvdontneed=1`.
* [devicedb] Upgraded DeviceDB from 1.9.2 to 1.9.4.
* [mbed-devicejs-bridge] Mapped new device function - accelerometer, gravity sensor, magnetometer, pressure, TVOC, gyroscope, signal strength, tap detection, CO2, step counter, Euler angles and heading to LwM2M objects and resources.
* [relay-term] Added websocket ping-pong handler for relay-term.
* [ble-pt] Added features to BLE protocol translator:
	* APIs to scan the gateway for BLE devices and report the MAC address, name and RSSI of all discovered devices.
	* APIs to allow users to dynamically onboard a BLE device found in the response of the above API.
	* Disabled the static whitelist of BLE service UUIDs.
	* Added support for Nordic Thingy and Embedded Planet Agora board.
* [os] Switched init system from SysVinit to SystemD.
* [os] Introduced `gai.conf` to control the sorting order of the addresses resolved by libc library. By default, IPv4 is preferred over IPv6.
* [os] Removed the cronjob, which periodically restarted the gateway services.

### Bug fixes
* Fixed the situation in which the maestro daemon can run out of the system resources and cause the platform to reboot.
* Fixed the issue of DHCP client being unable to renew the IP address lease.
* Fixed the intermittent failure of remote terminal not connecting to Pelion cloud.
* Fixed the intermittent bug in the maestro daemon, causing it not to restart a monitoring process that exits unexpectedly.

### Known issues
* While provisioning your gateway, please use `vendor-id=42fa7b48-1a65-43aa-890f-8c704daade54` to unlock the rich node features, such as gateway logs and gateway terminal in the Pelion web portal.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,6 @@
 scripts/config.txt
 scripts/interfaces
 scripts/wpa_supplicant.conf
+recipes-wigwag/mbed-edge-core/files/mbed_cloud_dev_credentials.c
+recipes-wigwag/mbed-edge-core/files/update_default_resources.c
+recipes-core/ww-console-image-initramfs-init/files/upgradeCA.cert
```
Comment (costanic) [inline .gitignore:8]: do we still need this file?  I think this should have been removed when verified boot feature was removed.

File: conf/bblayers.conf.sample [hunk 2]
```diff
@@ -1,20 +1,22 @@
...
-    ##OEROOT##/meta-openembedded/meta-oe \
-    ##OEROOT##/meta-openembedded/meta-filesystems \
-    ##OEROOT##/meta-openembedded/meta-networking \
-    ##OEROOT##/meta-openembedded/meta-perl \
-    ##OEROOT##/meta-openembedded/meta-python \
-    ##OEROOT##/meta-raspberrypi \
-    ##OEROOT##/meta-security \
-    ##OEROOT##/meta-pelion-edge \
-    ##OEROOT##/meta-nodejs \
+    ${OEROOT}/meta \
```
Comment (costanic) [inline conf/bblayers.conf.sample:11]: This change doesn't seem right.  Did Yocto change the way it parses the sample files?  See here: https://www.yoctoproject.org/docs/3.1.1/ref-manual/ref-manual.html#structure-build-conf-bblayers.conf

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,34 +1,63 @@
...
+### Bug fixes
+* Fixed the situation in which the maestro daemon can run out of the system resources and cause the platform to reboot.
+* Fixed the issue of DHCP client being unable to renew the IP address lease.
+* Fixed the intermittent failure of remote terminal not connecting to Pelion cloud.
+* Fixed the intermittent bug in the maestro daemon, causing it not to restart a monitoring process that exits unexpectedly.
+
+### Known issues
+* While provisioning your gateway, please use `vendor-id=42fa7b48-1a65-43aa-890f-8c704daade54` to unlock the rich node features, such as gateway logs and gateway terminal in the Pelion web portal.
+
+## 1.0.0
```
[the remark this answers is not in the document]
Comment (ygoyal18) [reply CHANGELOG.md:30]: This is referring to 1.0.0 release notes. 2.0.0 is defined above.

File: recipes-wigwag/wwrelay-utils/files/wwrelay [hunk 4]
```diff
@@ -0,0 +1,10 @@
+#! /bin/sh
+
+if [ ! -e /userdata/edge_gw_config/identity.json ] ; then
+  # This should not be needed, but hey maestro keeps crashing on me
+  # TODO: remove this if when maestro reliably gets an ip
```
[the remark this answers is not in the document]
Comment (ygoyal18) [reply recipes-wigwag/wwrelay-utils/files/wwrelay:5]: Ya, I haven't tested without it. I think we still need it.

File: recipes-wigwag/mbed-edge-examples/files/0001-disable-doxygen.patch [hunk 5]
```diff
@@ -1,3 +1,12 @@
+From 18fa2d72d994040be53cd2c5ccf7d7ba0aef9937 Mon Sep 17 00:00:00 2001
```
[the remark this answers is not in the document]
Comment (ygoyal18) [reply recipes-wigwag/mbed-edge-examples/files/0001-disable-doxygen.patch:1]: Ya still needed.

File: .gitignore (hunk 1 above)
Comment (ygoyal18) [reply .gitignore:8]: Sure, will remove it.

File: conf/bblayers.conf.sample (hunk 2 above)
Comment (ygoyal18) [reply conf/bblayers.conf.sample:11]: Ya I reverted the way it is used in 1.0.0. Raising a PR to bring this change.

**Comment by ygoyal18** [comment]
Wait for #152, #153

**Comment by ygoyal18** [comment]
also wait for #154

**Pull request merged**